app/src/online/java/com/kamron/pogoiv/updater/AppUpdateUtil.java

```diff
@@ -94,4 +103,15 @@ public void onClick(DialogInterface dialogInterface, int i) {
         AlertDialog dialog = builder.create();
         return dialog;
     }
+
+    public static void deletePreviousApkFile(Context context) {
+        if (ContextCompat.checkSelfPermission(context,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED){
+            String newApkFilePath = context.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS) + "/" + FILE_NAME;
```

Comment:
Actually, why isn't the delete in the Download service onStartCommand() not sufficient to delete this file? Isn't that service started all the time?

I'm not sure I understand the reason for splitting up the code into AppUpdateUtil and DownloadUpdateService, but is there any reason you consider that AppUpdateUtil is a better location for this code instead of DownloadUpdateService? Just a thought. I'm not sure what the right answer is to this part.